### ISSUE TITLE
[no-duplicates] issue with namespace imports getting ignored

### ISSUE THREAD

**Issue by Yukinosuke-Takada** [import-js/eslint-plugin-import#3204]
Following the sample code in [docs](https://github.com/import-js/eslint-plugin-import/blob/main/docs/rules/no-duplicates.md), the following code should give error:

```js
import SomeDefaultClass from './mod'
import * as names from './mod'
```

I'm not sure if this is the expected behavior, but either way, it doesn't match what's written in the documentation.

environment:
- package version:
  `eslint-plugin-import@2.32.0`, `eslint@9.31.0`, `node@24.2.0` 
- config:
  ``` 
  rules: {
    'import/no-duplicates': 'error',
  },
  ```

Additional info:
The following code gives these errors.
```js
import SomeDefaultClass from './mod'       // Error: './mod' imported multiple times. eslint(import/no-duplicates)
import SomeDefaultClass2 from './mod'      // Error: './mod' imported multiple times. eslint(import/no-duplicates)
import { something } from './mod'          // Error: './mod' imported multiple times. eslint(import/no-duplicates)
import { something2 } from './mod'         // Error: './mod' imported multiple times. eslint(import/no-duplicates)
import * as names from './mod'
```

**Comment by Yukinosuke-Takada** [comment]
Is `import SomeDefaultClass, * as names from './mod';` allowed in JS in the first place?
